File: LeetCode/443. String Compression/StringCompression.ts
function compress(chars: string[]): number {
  //chars = ["a","a","b","b","c","c","c"]
  //"a", "2", "b", "2", "c", "3"
    let pointerAindex = 0;
    let pointerBindex = 0;

    while(pointerBindex < chars.length){
        let counter = 0;
        console.log('what is pointerAindex before:', pointerAindex);

        let currentChar = chars[pointerBindex];
        console.log('what is currentChar:', currentChar);

        while(pointerBindex < chars.length && chars[pointerBindex] === chars[pointerAindex]){
            console.log("chars[pointerBindex] === chars[pointerAindex]");

            pointerBindex++;
            console.log('what is pointerBindex after:', pointerBindex);

            counter++
            console.log('what is counter:', counter);
        }

        chars[pointerAindex] = currentChar;
        console.log('what is chars[pointerAindex] after:', chars[pointerAindex]);
        console.log('what is chars after:', chars);

        pointerAindex++
        console.log('what is pointerAindex after:', pointerAindex);

        if(counter > 1){
            console.log("counter > 1");

            const strCount = counter.toString();
            console.log("what is strCount: ", strCount);

            for(let c of strCount){
                console.log('what is c:', c);

                chars[pointerAindex] = c;
                console.log("what is chars[pointerAindex] after: ", chars[pointerAindex]);
                console.log('what is chars after:', chars);

                pointerAindex++;
                console.log("what is pointerAindex after: ", pointerAindex);
            }
        }
    }
    return pointerAindex;
}

File: LeetCode/443. String Compression/StringCompression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LeetCode.443. String Compression
{
    public class StringCompression
{
        
        public int Compress(char[] chars) {
            int pointerA = 0;
            int pointerB = 0;

            while(pointerB < chars.Length){
                //Console.WriteLine("you're in the while loop!");
                int counter = 0;
                char currentChar = chars[pointerB];
                
                while(pointerB < chars.Length && chars[pointerB] == currentChar){
                    //Console.WriteLine("you're in the while loop!");
                    pointerB++;
                    counter++;
                }

                chars[pointerA] = currentChar;
                pointerA++;

                if(counter > 1){
                    string countStr = counter.ToString();
                    foreach(char num in countStr){
                        chars[pointerA] = num;
                        pointerA++;
                    }
                }
            }
            return pointerA;
        }
    }
}